DOC: MAINT: Add NEP 43 links to NEP 42

bjnath opened the pull request:
Now that NEP 43 is merged, adds the missing NEP 43 links to NEP 42.

bjnath committed:
DOC: MAINT: Add NEP 43 links to NEP 42

mattip commented:
Could these be
```
:ref:`NEP 43`
``` 
instead of 
```
:ref:`NEP 43 <NEP43>`
```

bjnath committed:
DOC: Include NEP titles for first reference in NEP 42

Rather than saying 'NEP 4x', the links
in the note at the top now include the NEP titles as well.

bjnath commented:
That doesn't work ("undefined label").

The link can be written <code>:ref:\`NEP43\`</code> (without the space) but then the NEP title is included.  We don't want the title every time, just "NEP 43", which is the reason for the current coding.

OTOH, for the note at the top of the page, including the NEP titles seems a good idea, so I made that change.

mattip commented:
We probably had the same discussion before, thanks for reminding me :+1:  The [rendered page](https://16388-908607-gh.circle-artifacts.com/0/doc/neps/_build/html/nep-0042-new-dtypes.html) looks good.

mattip commented:
Thanks @bjnath
